src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -522,22 +599,158 @@ test_small_initial_buffer (void *unused)
    _mongoc_host_list_destroy_all (rr_data.hosts);
 }
 
+bool
+_mock_resolver (const char *service,
+                mongoc_rr_type_t rr_type,
+                mongoc_rr_data_t *rr_data,
+                size_t initial_buffer_size,
+                bson_error_t *error)
+{
+   rr_data->count = 2;
+   rr_data->hosts = MAKE_HOSTS ("localhost.test.build.10gen.cc:27017",
+                                "localhost.test.build.10gen.cc:27018");
```

Comment:
Does this actually "mock the addition of the following DNS record" (quoting the [prose test](https://github.com/mongodb/specifications/blob/master/source/polling-srv-records-for-mongos-discovery/tests/README.rst))? I'm likely missing some context from the prose test, but where does the SRV record only contain "localhost.test.build.10gen.cc:27017" to begin with? Is that already specified by the DNS configuration that all drivers are depending on from the get-go?

If SRV polling isn't supposed to happen at all, why is this functional called at all? Wouldn't libmongoc do an initial resolution on `mongodb+srv://test3.test.build.10gen.cc` (against real DNS) and then never poll? In that case, should we be asserting that this function is actually never invoked?